src/spam-protection/spam-protection-action-creator.ts

```diff
@@ -0,0 +1,88 @@
+import { createAction, createErrorAction, ThunkAction } from '@bigcommerce/data-store';
+import { concat, empty, of, throwError, Observable, Observer } from 'rxjs';
+import { switchMap, take } from 'rxjs/operators';
+
+import { InternalCheckoutSelectors } from '../checkout';
+import { MissingDataError, MissingDataErrorType } from '../common/error/errors';
+
+import GoogleRecaptcha from './google-recaptcha';
+import { SpamProtectionAction, SpamProtectionActionType } from './spam-protection-actions';
+import { SpamProtectionOptions } from './spam-protection-options';
+import SpamProtectionRequestSender from './spam-protection-request-sender';
+
+export default class SpamProtectionActionCreator {
+    constructor(
+        private _googleRecaptcha: GoogleRecaptcha,
+        private _requestSender: SpamProtectionRequestSender
+    ) {}
+
+    initialize(options?: SpamProtectionOptions): ThunkAction<SpamProtectionAction, InternalCheckoutSelectors> {
+        return store => Observable.create((observer: Observer<SpamProtectionAction>) => {
+            const spamProtectionElementId = options ? options.containerId : 'spamProtectionContainer';
+
+            if (!options && !document.getElementById(spamProtectionElementId)) {
+                const spamProtectionElement = document.createElement('div');
+                spamProtectionElement.setAttribute('id', spamProtectionElementId);
+                document.body.append(spamProtectionElement);
+            }
+
+            const state = store.getState();
+            const config = state.config.getConfig();
+
+            if (!config) {
+                throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
```

Comment:
Perhaps this observable action can be written using pipe and operators instead? You can keep it in this form but you'll need to find a way to catch this error and transform it into an error action.